### PR TITLE
Add DeviantArt provider

### PR DESCRIPTION
**Pull request by tparnell8**

### CODE REVIEW

File: src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public class DeviantArtAuthenticationHandler : OAuthAuthenticationHandler<DeviantArtAuthenticationOptions> {
+        public DeviantArtAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+			request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: The indentation is weird :smile:

Comment (tparnell8) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: in what way?

**Comment by tparnell8** [comment]
So...I'm not sure why appveyor and travis says this won't build, as I have 2 different machines here that can both build this. 

I even did a fresh checkout and built just fine.

File: src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs (hunk 1 above)
Comment (tparnell8) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: Oh I see, gimmie one sec tabs vs spaces issue on my laptop vs my regular comp

**Comment by tparnell8** [comment]
could this be a problem with a newer sake package? 

>/home/travis/build/aspnet-contrib/AspNet.Security.OAuth.Providers/packages/KoreBuild/build/_k-standard-goals.shade(95,47): error CS0103: The name `IsTeamCity' does not exist in the current context

**Comment by PinpointTownes** [comment]
Thanks for this new contribution! I'll fix the build issue tomorrow and merge this PR ASAP.

/cc @jerriep

File: src/AspNet.Security.OAuth.Providers.DeviantArt/project.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿{
+    "version": "1.0.0-*",
+    "description": "ASP.NET 5 security middleware enabling DeviantArt authentication.",
+    "authors": [ "Tommy Parnell" ],
+    "owners": [ "Kévin Chalet", "Jerrie Pelser" ],
+
+    "projectUrl": "https://DeviantArt.com/aspnet-contrib/AspNet.Security.OAuth.Providers",
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/project.json]: Invalid URL fixed before merging.

File: AspNet.Security.OAuth.Providers.sln [hunk 3]
```diff
@@ -19,6 +19,8 @@ Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AspNet.Security.OAuth.WordP
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AspNet.Security.OAuth.Yahoo", "src\AspNet.Security.OAuth.Yahoo\AspNet.Security.OAuth.Yahoo.xproj", "{F79E8D15-F296-424C-A50E-AE39EB2C609F}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AspNet.Security.OAuth.DeviantArt", "src\AspNet.Security.OAuth.Providers.DeviantArt\AspNet.Security.OAuth.DeviantArt.xproj", "{9E09DCBC-20DB-4CD8-8A6D-5FA1958EEF14}"
```
Comment (PinpointTownes) [inline AspNet.Security.OAuth.Providers.sln]: Invalid project path fixed before merging.

**Comment by PinpointTownes** [comment]
Merged, thanks @tparnell8!

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/commit/eeafe4354d9fc26abea0eceff6e00152c010c8d0

**Pull request closed without merging**